src/iccpd/src/iccp_cli.c

```diff
@@ -208,8 +213,18 @@ int set_local_address(int mid, const char* addr)
 
     len = strlen(addr);
     memset(csm->sender_ip, 0, INET_ADDRSTRLEN);
+    if (len > INET_ADDRSTRLEN)
```

Comment:
for this one, did you confirm that the csm->sender_ip is created with the same "INET_ADDRSTRLEN" size, if not, you will need to change accordingly... if it is then it should be fine.

Reply:
yes, I checked